src/TensorFlowDiffEq.jl

```diff
@@ -14,6 +14,46 @@ odetf(;hl_width=64) = odetf(hl_width)
 
 export odetf
 
+
+immutable TensorFlowInterpolation{uElType}
+    sess::Session
+    u::Tensor{uElType}
+end
+
+
+grad_node(interp, deriv::Type{Val{0}}) = interp.u
+
+function grad_node{N}(interp, deriv::Type{Val{N}})
+    N >= 0 || throw(DomainError())
+    #GOLDPLATE: this code could be enhanced, a lot. Right now it creates new nodes in the graph every time it is called. Which is a minor waste. The fix makes the code pretty hard to read and so I do not think it is worth it.
+    as_default(interp.sess.graph) do
+        prev = grad_node(interp, Val{N-1})
+        tt = interp.sess.graph["tt"] #load the `tt` node from the graph
+        grads(prev, tt)
+    end
+end
+
+function (id::TensorFlowInterpolation){N}(tvals, idxs, deriv::Type{Val{N}})
+    gn = grad_node(id, derviv)
+    vals = run(id.sess, gn, Dict(t=>tvals))
+    #PREM-OPT: the indexing could be moved inside the network, to avoid even calculating gradients for columns are are not using, by first slicing `interp.u` inside the `grad_node` function.
+    vals[:,idxs]'
+end
+
+function (id::TensorFlowInterpolation){N}(v, tvals, idxs, deriv::Type{Val{N}})
+    # In-place version, noting that truely inplace operations between julia and tensorflow are actually impossible
```

Comment:
Though couldn't you `copy!` into a user defined array to bring things back from the GPU?